Lib/test/test_typing.py

```diff
@@ -838,6 +838,143 @@ class C(Generic[Unpack[Ts]]): pass
         self.assertNotEqual(C[Unpack[Ts1]], C[Unpack[Ts2]])
 
 
+class TypeVarTuplePicklingTests(BaseTestCase):
```

Comment:
Holy moly these are really detailed and comprehensive tests. I was expecting something more barebones.

You might want to also take a look at how other types test pickle. They need to cover all the pickle protocol versions.
https://github.com/python/cpython/blob/5bdeab67a609308b59f53e7b5f738269eaf28095/Lib/test/test_typing.py#L5018-L5031

Reply:
Ah, good point. I felt a bit uneasy about having loops in all these tests, so I've implemented it using a decorator. (I've also gone through and changed the other tests cases to use the decorator too, making the test bodies a bit simpler, but let me know if you'd prefer this done in a separate PR/not at all, and I'll revert that commit.)

Reply:
Thanks. The refactoring is good. But yeah I'd prefer them done in another PR please. There're some points of contention.

Reply:
Done.